### ISSUE TITLE
`path_startswith()` broken on Windows

### ISSUE THREAD

**Issue by mih** [datalad/datalad#2563]
```
[00:15:30] ======================================================================
[00:15:30] FAIL: datalad.tests.test_utils.test_path_startswith
[00:15:30] ----------------------------------------------------------------------
[00:15:30] Traceback (most recent call last):
[00:15:30]   File "C:\Miniconda35\envs\test-environment\lib\site-packages\nose\case.py", line 198, in runTest
[00:15:30]     self.test(*self.arg)
[00:15:30]   File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\test_utils.py", line 760, in test_path_startswith
[00:15:30]     ok_(path_startswith('/a/b', '/a'))
[00:15:30] AssertionError
```

**Comment by mih** [comment]
https://github.com/datalad/datalad-revolution uses pathlib to avoid this code entirely

**Comment by mih** [comment]
The release of 0.12 with this fix is imminent (#3993) closing.
